sw/host/opentitanlib/src/test_utils/poll.rs

```diff
@@ -0,0 +1,42 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+//! Helper functions for polling in tests.
+//!
+//! Useful for waiting for results that may not be available immediately.
+
+use std::cmp;
+use std::thread;
+use std::time::{Duration, Instant};
+
+use anyhow::bail;
+
+/// Poll some operation until it returns `true` or a timeout is hit.
+///
+/// The provided function should return `Some(x)` when the condition is met,
+/// or `None` to continue polling.
+///
+/// `delay` can be used to prevent retrying the operation too frequently.
+pub fn poll_until<F>(timeout: Duration, delay: Duration, f: F) -> anyhow::Result<()>
```

Comment:
Do we want to accept closures in the future? If so, F should implement `F:Fn()`

Reply:
I've specified `F: Fn() -> anyhow::Result<bool>` in the where clause below, is that what you mean?

```rust
pub fn poll_until<F>(timeout: Duration, delay: Duration, f: F) -> anyhow::Result<()>
where
    F: Fn() -> anyhow::Result<bool>,
{
```

Related, but I haven't read up on the difference between `Fn`, `FnMut`, and `FnOnce` so maybe this bound needs changing anyway.